span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/tenant/TenantIdProvider.java

```diff
@@ -6,9 +6,11 @@
 import org.slf4j.Logger;
 
 public interface TenantIdProvider {
+
   Optional<String> getTenantIdTagKey();
 
-  Optional<String> getTenantId(Map<String, JaegerSpanInternalModel.KeyValue> tags);
+  Optional<String> getTenantId(
```

Comment:
Why take the tags and the span? Seems like the span should replace the tags argument since it also includes the tags.

Reply:
The only caller of this method was constructing a map from the list of span tags. Since it avoids iterating over the whole list of tags again, I thought it would be better to use the map of tags.

Reply:
OK forgot about the list-> map conversion, but there's actually two different things going on here. The spans we're converting from a list to a map, but we're also normalizing the keys to be all lower cased.

1. We should be consistent with the behavior of the process, too. The process goes through the exact same logic later - keys lower cased and converted to map. So for the process we're both doing that repetition we were trying to avoid on tags and using the potentially pre-lower-cased keys.
2. Assuming we start taking in a map for both for consistency, the span arg can be dropped and the names should reflect that these are not raw but already normalized attribute pairs.

Reply:
Made this change.